libsolutil/Assertions.h

```diff
@@ -38,21 +40,41 @@ namespace solidity::util
 #define ETH_FUNC __func__
 #endif
 
-/// Assertion that throws an exception containing the given description if it is not met.
-/// Use it as assertThrow(1 == 1, ExceptionType, "Mathematics is wrong.");
-/// Do NOT supply an exception object as the second parameter.
-#define assertThrow(_condition, _exceptionType, _description) \
+namespace assertions
+{
+
+inline std::string stringOrDefault(std::string _string, std::string _defaultString)
+{
+	return (!_string.empty() ? _string : _defaultString);
+}
+
+}
+
+/// Base macro that can be used to implement assertion macros.
+/// Throws an exception containing the given description if the condition is not met.
+/// Allows you to provide the default description for the case where the user of your macro does
+/// not provide any.
+/// The second parameter must be an exception class (rather than an instance).
+#define assertThrowWithDefaultDescription(_condition, _exceptionType, _description, _defaultDescription) \
 	do \
 	{ \
 		if (!(_condition)) \
 			::boost::throw_exception( \
 				_exceptionType() << \
-				::solidity::util::errinfo_comment(_description) << \
+				::solidity::util::errinfo_comment( \
+					::solidity::util::assertions::stringOrDefault(_description, _defaultDescription) \
+				) << \
 				::boost::throw_function(ETH_FUNC) << \
 				::boost::throw_file(__FILE__) << \
 				::boost::throw_line(__LINE__) \
 			); \
 	} \
 	while (false)
 
+/// Assertion that throws an exception containing the given description if it is not met.
+/// Use it as assertThrow(1 == 1, ExceptionType, "Mathematics is wrong.");
+/// The second parameter must be an exception class (rather than an instance).
+#define assertThrow(_condition, _exceptionType, _description) \
+	assertThrowWithDefaultDescription(_condition, _exceptionType, _description, "Assertion failed")
```

Comment:
Is `assertThrow()` really an assertion? In most cases I've seen it's just used as a convenience helper to throw a boost exception.